docs/writing-benchmarks.md

```diff
@@ -0,0 +1,78 @@
+
+
+#### @Benchmark
+
+The `@Benchmark` annotation is used to specify the methods that you want to measure the performance of. Basically, it's the actual test you're running. The code you want to benchmark goes inside this method. In our example, the `benchmarkMethod` function is annotated with `@Benchmark`, which means the toolkit will measure the performance of the operation of summing all the integers in the list.
+
+#### @TearDown
+
+The `@TearDown` annotation is used to denote a method that's executed after the benchmarking method(s), typically responsible for cleaning up or deallocating any resources or conditions that were initialized or consumed during the benchmark. For instance, if your benchmark creates temporary files or opens network connections, the method marked with `@TearDown` is where you would put the code to delete those files or close those connections. The `@TearDown` annotation helps you avoid performance bias and ensure the proper maintenance of resources and the preparation of a clean environment for the next run. In our example, the `cleanup` function annotated with `@TearDown` is used to clear the list of integers after each benchmark iteration, ensuring that each test starts with the same initial state.
+
+#### @BenchmarkMode
+
+The `@BenchmarkMode` annotation sets the mode of operation for the benchmark. Applying the `@BenchmarkMode` annotation requires specifying a mode from the `Mode` enum, which includes several options. `Mode.Throughput` measures the raw throughput of your code in terms of the number of operations it can perform per unit of time, such as operations per second. When this mode is selected, the toolkit runs your code in a loop for a set duration, then calculates and reports the operation throughput. Conversely, `Mode.AverageTime` is used when you're more interested in the average time it takes to execute an operation. Here, the toolkit runs your code multiple times, records the time each operation takes, and calculates and reports the average time. Without an explicit `@BenchmarkMode` annotation, the toolkit defaults to `Mode.Throughput`. In our example, `@BenchmarkMode(Mode.Throughput)` is used, meaning the benchmark focuses on the number of times the benchmark method can be executed per unit of time.
```

Comment:
Could you please clarify how you inferred the information about how the toolkit runs in different modes?

Reply:
In attempt to provide a quality annotation description, I left oversimplifications (not based on code) about how the toolkits operates. I have removed them in the latest commit.